nixos/modules/services/web-apps/stash.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.stash;
+in
+{
+  options = {
+    services.stash = {
+      enable = mkEnableOption "An organizer for your porn, written in Go.";
+
+      package = mkPackageOption pkgs "stash" { };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/stash";
+        description = "The state directory where stash stores its metadata, cache and generated files.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "User account under which stash runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "Group under which stash runs.";
+      };
```

Comment:
Assuming `dataDir` is only used for the three mentioned purposes, I would recommend replacing it with always using `StateDirectory`. That would make it easier to also remove the `user` and `group`, and use `DynamicUser` instead.